project/searching.py

```diff
@@ -15,16 +16,31 @@ def binary_search(arr, target):
   low = 0
   high = len(arr)-1
 
-  # TO-DO: add missing code
+  while len(arr) >= 1:
+    mid = int((high - low) / 2)
+
+    if target == arr[mid]:
+      return mid
+    elif target < arr[mid]:
+      high = mid
+    else:
+      low = mid
 
   return -1 # not found
 
 
 # STRETCH: write a recursive implementation of Binary Search 
 def binary_search_recursive(arr, target, low, high):
   
-  middle = (low+high)/2
+  middle = int((high - low) / 2)
 
```

Comment:
I like that you used middle = int((high - low) / 2) for the binary search stretch